webtop/__init__.py

```diff
@@ -136,60 +140,47 @@ def render_stats(stats: dict, _format: str) -> str:
     return output
 
 
-def main() -> None:
+async def main() -> None:
     args = parse_args()
     assert are_args_valid(args)
 
     results: Deque[Result] = deque(maxlen=args.request_history)
-    results_lock = Lock()
     shutdown_event = Event()
 
-    def shutdown_signal_handler(signum, frame):
+    def shutdown_signal_handler(_, __):
         shutdown_event.set()
 
     for shutdown_signal in (signal.SIGINT, signal.SIGTERM):
         signal.signal(shutdown_signal, shutdown_signal_handler)
 
-    threads = []
-
-    def start_daemon(target: Callable[[], Any]) -> None:
-        # Fly, Pantalaimon!
-        thread = Thread(target=target)
-        threads.append(thread)
-        thread.daemon = True
-        thread.start()
+    tasks = []
 
-    def renderer() -> None:
+    async def renderer() -> None:
         while True:
             if shutdown_event.is_set():
                 return
 
-            with results_lock:
-                stats = build_stats(url=args.url, method=args.method, results=results)
+            stats = build_stats(url=args.url, method=args.method, results=results)
             output = render_stats(stats, _format=args.output_format)
 
             os.system("clear")
             print(output, flush=True)
 
-            sleep(0.1)
+            await asyncio.sleep(0.1)
 
-    start_daemon(target=renderer)
+    tasks.append(renderer())
+    timeout = aiohttp.ClientTimeout(connect=args.timeout)
+    connector = aiohttp.TCPConnector(force_close=True, limit=0)
+    async with aiohttp.ClientSession(timeout=timeout, connector=connector) as session:
 
-    def worker() -> None:
-        while True:
-            result = request(url=args.url, method=args.method, timeout=args.timeout)
-            with results_lock:
+        async def worker() -> None:
+            while not shutdown_event.is_set():
+                result = await request(url=args.url, method=args.method, session=session)
                 results.append(result)
 
-            if shutdown_event.is_set():
-                return
-
-    for i in range(args.threads):
-        start_daemon(target=worker)
-
-    for thread in threads:
-        thread.join()
+        tasks.extend([worker()] * args.workers)
```

Comment:
I think this is actually adding the same worker multiple times

Reply:
Try:

```
for i in range(0, args.workers):
    tasks.append(worker())
```

Reply:
PFFT NERD

Reply:
k, 1 sec

Reply:
Using `_` for the variable name because convention, and `range` defaults to starting at 0.